jme3-core/src/main/java/com/jme3/scene/Geometry.java

```diff
@@ -601,6 +601,39 @@ public void setMorphState(float[] state) {
         this.dirtyMorph = true;
     }
 
+    /**
+     * Set the state of the morph with the given name.
+     * 
+     * If the name of the morph is not found, no state will be set.
+     * 
+     * @param morphTarget The name of the morph to set the state of
+     * @param state The state to set the morph to
+     */
+    public void setMorphState(String morphTarget, float state) {
+        if (mesh == null || mesh.getMorphTargets().length == 0) {
```

Comment:
You can use `!mesh.hasMorphTargets()` instead of `mesh.getMorphTargets().length == 0`